cluster-autoscaler/core/scaledown/legacy/legacy.go

```diff
@@ -1018,7 +1008,7 @@ func (sd *ScaleDown) scheduleDeleteEmptyNodes(emptyNodesToRemove []simulator.Nod
 					sd.context.LogRecorder.Eventf(apiv1.EventTypeNormal, "ScaleDownEmpty", "Scale-down: empty node %s removed", nodeToDelete.Name)
 				}
 			}()
-			if err := evictDaemonSetPods(sd.context.ClusterSnapshot, nodeToDelete, client, sd.context.MaxGracefulTerminationSec, time.Now(), DaemonSetEvictionEmptyNodeTimeout, DeamonSetTimeBetweenEvictionRetries, recorder, evictByDefault); err != nil {
+			if err := actuation.EvictDaemonSetPods(sd.context, nodeToDelete, time.Now(), actuation.DaemonSetEvictionEmptyNodeTimeout, actuation.DeamonSetTimeBetweenEvictionRetries); err != nil {
```

Comment:
nit: if we had some "Evictor" object, time constants could be stored as its members instead of being passed here, reducing the number of parameters. Since the context pointer doesn't change, it could be moved there as well.

Reply:
Yeah good idea, that's much better. I put it into Actuator for now though, since it's only used there. Otherwise it'd have to be extracted to a separate package to avoid an import cycle between actuation and context. That should be easy enough to do if it's needed elsewhere in the future.